src/cpp/r/RSexp.cpp

```diff
@@ -838,6 +838,77 @@ bool inherits(SEXP object, const char* S3Class)
    return Rf_inherits(object, S3Class);
 }
 
+std::set<std::string> makeNseFunctions()
+{
+   std::set<std::string> nseFunctions;
+   nseFunctions.insert("quote");
+   nseFunctions.insert("substitute");
+   nseFunctions.insert("match.call");
+   nseFunctions.insert("library");
+   nseFunctions.insert("require");
+   nseFunctions.insert("enquote");
+   nseFunctions.insert("expression");
+   nseFunctions.insert("evalq");
+   nseFunctions.insert("subset");
```

Comment:
`with`, `within`, `transform`, ... I could come up with a more substantive list, but I think you're missing a lot